### PR TITLE
Ember model modal

### PR DESCRIPTION
**Pull request by jweakley**
Adds a prototype modal.

### CODE REVIEW

File: tests/integration/components/modals/-body-test.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+
+module('Integration | Component | modals/ body', function(hooks) {
```
Comment (harokevin) [inline tests/integration/components/modals/-body-test.js]: I don't quite care  for the formatting on `Integration | Component | modals/ body` but it is not that big of a deal.

I like the `Integration | Component | modals | body` format.

**Comment by harokevin** [comment]
Also, I spun up the app and the modal was not displaying on freestyle.

File: addon/components/bootstrap/edit-form.js [hunk 2]
```diff
@@ -38,16 +38,22 @@ export default Component.extend({
 
   asyncSaveTask: task(function * (asyncTask) {
       if (asyncTask) {
-        yield asyncTask();
+        if (yield asyncTask()) {
```
[the remark this answers is not in the document]
Comment (jweakley) [reply addon/components/bootstrap/edit-form.js]: This should probably just assume this worked unless it returns false.

File: addon/components/modals/ember-model.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  isNewModel: alias('model.isNew'),
+
+  saveTask: task(function * (beforeSave, afterSave) {
+    if (isPresent(beforeSave)) {
+      const beforeSaveResponse = yield beforeSave();
+      if (!beforeSaveResponse) {
+        return false;
+      }
+    }
+    const model = yield this.get('model');
```
[the remark this answers is not in the document]
Comment (jweakley) [reply addon/components/modals/ember-model.js]: The usage of model is not in the documentation. I have made https://github.com/wildland/ember-bootstrap-controls/issues/261 to follow up once we figure out what post freestyle looks like.

File: addon/components/progress-bars/-bar.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  role: 'progressbar',
+
+  minimumValue: 0,
+  maximumValue: 100,
+  currentValue: 100,
+
+  updateStyles() {
+    const max = this.get('maximumValue');
+    const min = this.get('minimumValue');
+    const total = max - min;
```
[the remark this answers is not in the document]
Comment (jweakley) [reply addon/components/progress-bars/-bar.js]: I added a follow up issue: https://github.com/wildland/ember-bootstrap-controls/issues/262

File: addon/components/bootstrap/edit-form.js (hunk 2 above)
Comment (harokevin) [inline addon/components/bootstrap/edit-form.js]:
```suggestion
        const response = yield asyncTask();
        const wasSuccessful = response || response === undefined;
        if (wasSuccessful) {
```

**Comment by harokevin** [comment]
Still can't see the modal locally.

**Comment by harokevin** [comment]
Recap of talking with Joe about this PR.

The `BOOTSTRAP SIMPLE MODAL` freestyle section does not work and should be removed.
This pr introduces 3 modals. A basic modal and a basic modal with minor enhancements. Both of these modals were build to be a private component for the only modal we intend to expose, the model modal.

Because the model-modal needs ember data to properly render in freestyle I am not requiring that it be functional in freestyle but I would like to require that:
- [ ] There is a nav item for model modal
- [ ] There is a description that highlights the use case of this component and the fact that ember data is a dependency
- [ ] There is a single advanced use case snippet for using the component that does not have to be functional, just displayed.

**Comment by Samsinite** [comment]
Is an ember-data model not something we can mock so that it works in freestyle?

[signature_2058582801]

Samuel Clopton | Chief Architect
wild.land<https://wild.land/>
sam@wild.land<mailto:sam@wild.land>
C: 509.308.2999

**Pull request merged**